libs/user_extensions/src/TTAlpsHistogramFiller.cpp

```diff
@@ -231,6 +231,93 @@ void TTAlpsHistogramFiller::FillJetHistograms(const shared_ptr<Event> event){
   }
 }
 
+void TTAlpsHistogramFiller::FillLooseDSAMuonsHistograms(const shared_ptr<Event> event){
+  float weight = GetEventWeight(event);
+  auto looseDSAMuons = event->GetCollection("LooseDSAMuons");
+
+  float nLooseDSAMuons = looseDSAMuons->size();
+  histogramsHandler->Fill("Event_nLooseDSAMuons", nLooseDSAMuons, weight);
+
+  for(auto dsaMuonObj : *looseDSAMuons){
+    float dxy = dsaMuonObj->Get("dxyPV");
+    float dz = dsaMuonObj->Get("dzPV");
+    float pt = dsaMuonObj->Get("pt");
+    float eta = dsaMuonObj->Get("eta");
+
+    histogramsHandler->Fill("LooseDSAMuons_dxy", dxy, weight);
+    histogramsHandler->Fill("LooseDSAMuons_dz", dz, weight);
+    histogramsHandler->Fill("LooseDSAMuons_pt", pt, weight);
+    histogramsHandler->Fill("LooseDSAMuons_eta", eta, weight);
+  }
+
+
+}
+
+void TTAlpsHistogramFiller::FillAllLooseMuonsHistograms(const shared_ptr<Event> event){
+  float weight = GetEventWeight(event);
+  auto looseMuons = event->GetCollection("LooseMuons");
+  auto looseDsaMuons = event->GetCollection("LooseDSAMuons");
+
+  PhysicsObjects allLooseMuons;
+
+  float min_deltaR = 99999;
+  float pt, eta, dxy, dz, dxyPV, dzPV;
+
+  for(auto muonObj : *looseMuons){
```

Comment:
I think that here you should do asNanoEvent(event) and use GetAllMuons method (e.g. with 0.01 delta R, if that's what you want). Then you can greatly simplify this method.

Reply:
Is there a Get function for the variables with the NanoEvent? I thought so but it doesn't seem to work for me

Reply:
Yes, it's [here](https://github.com/jniedzie/tea/blob/fdae016586b541a0edf196cf08c1a8d0ac846796/libs/extensions/include/NanoEvent.hpp#L11). And to get "replaced" muon (we should agree on some name and change it in the code) you can use [this method](https://github.com/jniedzie/tea/blob/fdae016586b541a0edf196cf08c1a8d0ac846796/libs/extensions/include/NanoEvent.hpp#L20).